relations/controllers.py

```diff
@@ -22,4 +27,232 @@ def service_status(params: MultiDict) -> Response:
 
     Returns ``200 OK`` if the service is up and ready to handle requests.
     """
-    return {'iam': 'ok'}, HTTPStatus.OK, {}
\ No newline at end of file
+    return {'iam': 'ok'}, HTTPStatus.OK, {}
+
+
+def create_new(arxiv_id_str: str,
```

Comment:
We try to leave the request payload interpretation up to the controllers as much as possible, rather than having the routes worry about the semantics of the payload. So the controller might instead accept any URL parameters (e.g. ``arxiv_id``, ``arxiv_ver``), and then a struct with the deserialized payload, e.g. ``Dict[str, Any]``. Would look something like:


```python
# routes.py
from flask import request, Response

@blueprint.route('/{arxiv_id}v{int:version}/relations', methods=['POST'])
def create(arxiv_id: str, version: int) -> Response:
    payload = request.get_json(force=True)    # Route handles deserialization.
    response_data, code, headers = controllers.create_new(arxiv_id, version, payload)
    ...

# controllers.py
Response = Tuple[Dict[str, Any], HTTPStatus, Dict[str, str]]

def create_new(arxiv_id: str, arxiv_ver: int, payload: Dict[str, Any]) -> Response:
    ...
```

Reply:
This is what I was thinking but not able to make a choice. I'll change the design.

Reply:
Cool. I also updated the [service design docs](https://arxiv.github.io/arxiv-arxitecture/crosscutting/services.html#routes) to clarify this